src/xclim/sdba/base.py

```diff
@@ -196,8 +198,7 @@ def get_coordinate(self, ds: xr.Dataset | None = None) -> xr.DataArray:
             if ds is not None:
                 cal = get_calendar(ds, dim=self.dim)
                 mdoy = max(
-                    xr.coding.calendar_ops._days_in_year(yr, cal)
-                    for yr in np.unique(ds[self.dim].dt.year)
+                    xr.coding.calendar_ops._days_in_year(ds[self.dim].dt.year, cal)
```

Comment:
Indeed, this fixes it, but it has the disadvantage that it continues to use a private xarray function. The solution in #2038 is independent of xarray, even if not as fancy. The main difference is in the case where the passed dataset only covers non-leap years (and has a standard calendar). Do you think that case is probable ?